src/queries/comment/circle/discussion.ts

```diff
@@ -1,16 +1,42 @@
-import { COMMENT_STATE, COMMENT_TYPE } from 'common/enums'
+import {
+  COMMENT_STATE,
+  COMMENT_TYPE,
+  PRICE_STATE,
+  SUBSCRIPTION_STATE,
+} from 'common/enums'
 import { connectionFromArray, cursorToIndex } from 'common/utils'
 import { CircleToDiscussionResolver } from 'definitions'
 
 const resolver: CircleToDiscussionResolver = async (
   { id },
   { input },
-  { dataSources: { atomService } }
+  { viewer, dataSources: { atomService }, knex }
 ) => {
   if (!id) {
     return connectionFromArray([], input)
   }
 
+  const records = await knex
```

Comment:
Would it be more consistent to use knex in `atomService`?

Reply:
It seems hard to use `atomService` since we use `join` here, but there are many places that have circle member query, maybe we should consider how to reuse it.

Reply:
It's quite different and difficult to simulate Prisma-like `join` feature using Knex-like functions. So, complex operations currently are not implemented in atomService. To me, it's fine to use `knex` here, and I believed it will replace by Prisma's way in the future.

Reply:
Just moved to `paymentService` for reusability.